lib/fs/basicfs.go

```diff
@@ -88,38 +88,22 @@ func (f *BasicFilesystem) rooted(rel string) (string, error) {
 		expectedPrefix += pathSep
 	}
 
-	// The relative path should be clean from internal dotdots and similar
-	// funkyness.
-	rel = filepath.FromSlash(rel)
-	if filepath.Clean(rel) != rel {
```

Comment:
@AudriusButkevicius  Is there a specific reason to be so strict about path arguments being "clean" in the fs package?  
I implemented `fs.Canonicalize` as the "central path-cleanup-function" e.g. to be used for `model.unifySubs` i.e. on external, potentially non-clean paths. Therefore I removed this check that `filepath.Clean` must have no effect.

Reply:
Without this check, what happens with paths like `foo/../../../bar`?

Reply:
Becomes `../../bar` and gets rejected, tested here
https://github.com/syncthing/syncthing/pull/4800/files#diff-fdedb2fe92a10afee3525356a2e710aeR75
and here
https://github.com/syncthing/syncthing/pull/4800/files#diff-aa4a1cf97526cef7590736f637040b8fR385

Reply:
So are you saying the check is a no-op, and we might as well keep it?

Reply:
No, I am saying that if we keep the check, then something like `fs.Canonicalize(./foo)` or `fs.Lstat("./foo")` will return an error. What I want and already do in the new `Canonicalize` function is to replace the check with just `file = filepath.Clean(file)`, i.e. for the example `fs.Canonicalize(./foo) == "foo"`.

Reply:
Right I might not have read well enough. As long as the clean happens I don’t think it’s critical that it was already clean when entering the function.